recipe/build-r-tiledbsoma.sh

```diff
@@ -20,4 +20,4 @@ fi
 
 export CXX17FLAGS="-Wno-deprecated-declarations -Wno-deprecated"
 
-${R} CMD INSTALL --build . ${R_ARGS}
+${R} CMD INSTALL --build --no-help . ${R_ARGS}
```

Comment:
@eddelbuettel @aaronwolen @mojaveazure may have thoughts here ... from my own (relatively R-amateur) understanding I tend to use:

```
alias rcb='rm -f *.tar.gz; R CMD build --no-build-vignettes --no-manual .'
alias rci='R CMD INSTALL --no-test-load --no-docs --no-html *tar.gz'
```

for interactive use ...

Reply:
I fear that if that is in regards to the build issue the help files are likely just the messenger you both may be aiming and firing at the messenger.

The help files have, as we are so fancy, a couple of call-outs to an `\Sexpr{}` leading to the loading of the package to evaluate the expression.  So we "just" fail to load the package earlier than we would at the next step.

I think this is symptomatic of an rpath issue.  We have one on non-Conda licked.  Are there directories that are different from non-Conda builds that we need to set an rpath for?

Reply:
To clarify, adding `--no-help` was a temporary workaround to trigger a different error message later in the package installation process. To my surprise, the [installation succeeded on osx-arm64](https://dev.azure.com/jdblischak/feedstock-builds/_build/results?buildId=278&view=logs&j=ced5d8de-8227-5f3f-33a1-45cf1592c45a&t=f7bcb349-6dca-53ba-d88d-1d9963e119ba&l=3719) when the help file installation step was skipped. However, the test section is skipped for cross-compiled builds, so I'm still somewhat skeptical that the package is functional, but it's a positive sign

```
+ cd apis/r
+ export DISABLE_AUTOBREW=1
+ DISABLE_AUTOBREW=1
+ [[ osx-arm64 == osx-64 ]]
+ export 'CXX17FLAGS=-Wno-deprecated-declarations -Wno-deprecated'
+ CXX17FLAGS='-Wno-deprecated-declarations -Wno-deprecated'
+ /Users/runner/miniforge3/conda-bld/tiledbsoma_1684260630295/_build_env/bin/R CMD INSTALL --build --no-help . --library=/Users/runner/miniforge3/conda-bld/tiledbsoma_1684260630295/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/R/library --no-test-load
* installing *source* package ‘tiledbsoma’ ...
** using staged installation
Error: No 'cmake' binary found
Execution halted
./configure: line 44: tools/build_libtiledbsoma.sh: No such file or directory
** libs
if [ `uname -s` = 'Darwin' ] && [ -f tiledbsoma.so ]; then \
    if [ -f ../inst/tiledb/lib/libtiledb.dylib ] ; then \
        install_name_tool -change libz.1.dylib @rpath/libz.1.dylib ../inst/tiledb/lib/libtiledb.dylib; \
        install_name_tool -add_rpath @loader_path/../tiledb/lib tiledbsoma.so; \
        install_name_tool -add_rpath @loader_path/../tiledbsoma/lib tiledbsoma.so; \
    fi; \
    if [ -f /usr/local/lib/libtiledb.dylib -a -f /usr/local/lib/libtiledbsoma.dylib ] ; then \
        install_name_tool -add_rpath /usr/local/lib tiledbsoma.so; \
** checking absolute paths in shared objects and dynamic libraries
* creating tarball
packaged installation of ‘tiledbsoma’ as ‘tiledbsoma_0.0.0.9024.tgz’
* DONE (tiledbsoma)
```

Reply:
I was able to remove the `cmake` not found error by installing `pkg-config` as a build dependency. However, that broke the osx-64 (ie amd64) build. I've seen this error about `Library not loaded: @rpath/R/lib/libR.dylib` before. It's either spurious or related to some of the upstream changes you've been making. I have some ideas to try

```
** R
** inst
** byte-compile and prepare package for lazy loading
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
Error: package or namespace load failed for ‘tiledbsoma’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/Users/runner/miniforge3/conda-bld/tiledbsoma_1684266988093/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/R/library/00LOCK-r/00new/tiledbsoma/libs/tiledbsoma.dylib':
  dlopen(/Users/runner/miniforge3/conda-bld/tiledbsoma_1684266988093/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/R/library/00LOCK-r/00new/tiledbsoma/libs/tiledbsoma.dylib, 6): Library not loaded: @rpath/R/lib/libR.dylib
  Referenced from: /Users/runner/miniforge3/conda-bld/tiledbsoma_1684266988093/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/Users/runner/miniforge3/conda-bld/tiledbsoma_1684266988093/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/R/library/tiledbsoma’
```

Reply:
(Scratching my head:  libR.dylib is what we get, that's from what would be `r-base` in Ubuntu. Conda not loading its own package? Hm.)

Reply:
My current hunch is that the calls to `install_name_tool` are messing with the standard conda `rpath` settings, which is why things break when trying to load it from a different location. But I have a plan to explicitly test this